### ISSUE TITLE
Implement Debug View

### ISSUE THREAD

**Issue by georgwiese** [scalableminds/chatroom#27]
It would be great to have a UI with the chatbot's state. Debugging chatbots on the command line is not fun, because debug information is printed in a rather unstructured way and gets mixed with messages.

A separate debug view would solve this.

# Basic Implementation

I hacked together a basic variant like this:

## Server

I added a new route to get the current tracker, including events:

```python


    @app.route("/conversations/<cid>/tracker", methods=["GET"])
    @check_cors
    def tracker(self, request, cid):

        tracker = self.agent.tracker_store.get_or_create_tracker(cid)
        tracker_state = tracker.current_state(should_include_events=True,
                                              only_events_after_latest_restart=True)

        request.setHeader("Content-Type", "application/json")
        return json.dumps(tracker_state)
```

An alternative would be to add the tracker to the `/log` response.

## JS Client

The JS frontend just polls the tracker and pretty-prints the important elements:

```javascript


      async function updateTrackerView() {
        var sessionId = window.sessionStorage.getItem("simple-chatroom-cid");
        var response = await fetch("http://localhost:5002/conversations/" + sessionId + "/tracker");
        var tracker = await response.json();

        // Remove Intent Ranking
        var {intent_ranking, ...latest_message} = tracker.latest_message;

        var html = "<h1>Slots</h1>" +
                   "<pre>" + JSON.stringify(tracker.slots, null, 2) + "</pre>" +
                   "<h1>Latest Message</h1>" +
                   "<pre>" + JSON.stringify(latest_message, null, 2) + "</pre>" +
                   "<h1>Events</h1>" +
                   "<pre>" + JSON.stringify(tracker.events, null, 2) + "</pre>";

        document.getElementById("tracker-view").innerHTML = html;
      }

      setInterval(updateTrackerView, 1000)
```

# Further Ideas

- We could also prominently desplay the last executed action. Would have to be searched for in the events.
- We should probably reverse events to have the most recent on top.
- We should probably also filter `intent_ranking` from the events as it is very verbose. Or at least make it collabsible.

**Comment by ctrado18** [comment]
I like this! Thanks! I am rather ne to JS. How would I implement use this JS script? You just wrote the main part but how do I run this at the end of the day?

**Comment by jstriebel** [comment]
fixed by #38
